examples/csv_read.rs

```diff
@@ -0,0 +1,36 @@
+use arrow2::error::Result;
+use arrow2::io::csv::read;
+use arrow2::record_batch::RecordBatch;
+
+fn read_path(path: &str, projection: Option<&[usize]>) -> Result<RecordBatch> {
+    // Create a CSV reader. This is typically created on the thread that reads from the file.
+    let mut reader = read::ReaderBuilder::new().from_path(path)?;
+
+    // initializes a default parser. A parser is stateless; it is just an API to parse rows.
+    // it must match the corresponding "inferer", that is used to infer the schema (if applicable).
+    let parser = read::DefaultParser::default();
+
+    // initializes a default parser. A parser is stateless; it is just an API to parse rows.
+    // it must match the corresponding "inferer", that is used to infer the schema (if applicable).
+    let schema = read::infer_schema(&mut reader, None, true, &read::infer)?;
+
+    // skip 0 (excluding the header) and read up to 100 entries
+    // this is IO-intensive and performs minimal CPU work. In particular,
+    // no deserialization is performed
+    let rows = read::read_rows(&mut reader, 0, 100)?;
```

Comment:
is this function the one that can be run in multiple threads? 
could it be run with async?

Reply:
I pushed a new example that uses multi-threading to describe the idea. 👍 

The CSV crate does not support async which means that we also do not :(